JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/videoPlayer/components/VideoPlayerOverlay.kt

```diff
@@ -0,0 +1,153 @@
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.jetstream.presentation.screens.videoPlayer.components
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.animation.slideInVertically
+import androidx.compose.animation.slideOutVertically
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.google.jetstream.presentation.theme.JetStreamTheme
+
+/**
+ * Handles the visibility and animation of the controls.
+ */
+@Composable
+fun VideoPlayerOverlay(
+    modifier: Modifier = Modifier,
+    state: VideoPlayerState = rememberVideoPlayerState(),
+    focusRequester: FocusRequester = remember { FocusRequester() },
+    isPlaying: Boolean,
+    centerButton: @Composable () -> Unit = {},
+    subtitles: @Composable () -> Unit = {},
+    controls: @Composable () -> Unit = {}
+) {
+    LaunchedEffect(state.controlsVisible) {
+        if (state.controlsVisible) {
+            focusRequester.requestFocus()
+        }
+    }
+
+    LaunchedEffect(isPlaying) {
+        if (!isPlaying) {
+            state.showControls(seconds = Int.MAX_VALUE)
+        } else {
+            state.showControls()
+        }
+    }
+
+    Box(
+        modifier = modifier.fillMaxSize(),
+        contentAlignment = Alignment.Center
+    ) {
+        AnimatedVisibility(state.controlsVisible, modifier, fadeIn(), fadeOut()) {
+            CinematicBackground(Modifier.fillMaxSize())
+        }
+
+        Column {
+            Box(Modifier.weight(1f),
+                contentAlignment = Alignment.BottomCenter) {
+                subtitles()
+            }
+
+            AnimatedVisibility(
+                state.controlsVisible,
+                modifier,
```

Comment:
You're sharing the same modifier among multiple children composables. Is this what you anticipated?